packages/ovee/src/dom/EventDelegate.ts

```diff
@@ -3,143 +3,153 @@ import Component from 'src/core/Component';
-            });
-        });
-    }
-
-    off(events: string, callback: Callback<Context>): void;
-    off(events: string, selector: string, callback: Callback<Context>): void;
-    off(events: string, target: Element, callback: Callback<Context>): void;
-    off(events: string, target: Element, selector: string, callback: Callback<Context>): void;
-    off(
-        _events: string,
-        _target: Callback<Context> | string | Element,
-        _selector?: Callback<Context> | string,
-        _callback?: Callback<Context>
-    ): void {
-        const {
-            events, target, selector, callback
-        } = this.getCastedParams(_events, _target, _selector, _callback);
-
-        events.split(' ').forEach((event) => {
-            this.listeners.forEach((item, index) => {
-                if (item.event === event && item.target === target
-                    && item.selector === selector && item.callback === callback) {
-                    target.removeEventListener(event, item.handler);
-                    this.listeners.splice(index, 1);
-                }
-            });
-        });
-    }
-
-    private getCastedParams(
-        events: string,
-        target: Callback<Context> | string | Element,
-        selector?: Callback<Context> | string,
-        callback?: Callback<Context>
-    ) {
-        const out: Partial<CastedParams<Context>> = { events, callback };
-
-        if (callback === undefined) {
-            if (selector === undefined) {
-                out.callback = target as Callback<Context>;
-                out.target = this.targetElement;
-            } else if (typeof target === 'string') {
-                out.callback = selector as Callback<Context>;
-                out.selector = target;
-                out.target = this.targetElement;
-            } else {
-                out.callback = selector as Callback<Context>;
-                out.target = target as Element;
-            }
-        } else {
-            out.target = target as Element;
-            out.selector = selector as string;
-        }
-
-        return out as CastedParams<Context>;
-    }
-
-    emit<D = any>(eventDesc: EventDesc, detail?: D): void {
-        emitEvent(this.targetElement, eventDesc, detail);
-    }
-
-    destroy(): void {
-        this.listeners = this.listeners.filter((item) => {
-            item.target.removeEventListener(item.event, item.handler);
-            return false;
-        });
-    }
+	listeners: Listener<Context>[] = [];
+	targetElement: Element;
+	context: Context;
+
+	constructor(targetElement: Element, context: Context) {
+		this.targetElement = targetElement;
+		this.context = context;
+	}
+
+	on(events: string, callback: Callback<Context>): void;
+	on(events: string, selector: string, callback: Callback<Context>): void;
+	on(events: string, target: Element, callback: Callback<Context>): void;
+	on(events: string, target: Element, selector: string, callback: Callback<Context>): void;
```

Comment:
I've recently realized that we are missing an option for hooking a passive listener

Reply:
what do you mean by a passive listener?

Reply:
`addEventListener('foobar', handler, { passive: true })`

Reply:
Added issue for that with proposed new API: #18

Reply:
Let's keep it for 2.2